pyccel/complexity/basic.py

```diff
@@ -25,16 +38,230 @@ def __init__(self, filename_or_text):
         settings = {}
         self._ast = pyccel.annotate(**settings).ast
 
+        # ...
+        functions = OrderedDict()
+        if pyccel.namespace.functions:
+            functions = pyccel.namespace.functions
+
+        for son in pyccel.sons:
+            functions.update(son.namespace.functions)
+
+        self._functions = functions
+        # ...
+
+        self._costs = OrderedDict()
+        self._shapes = OrderedDict()
+        self._symbol_map = {}
+        self._used_names = set()
+        self._visual = True
+        self._mode = None
+
     @property
     def ast(self):
         """Returns the Abstract Syntax Tree."""
         return self._ast
 
-    def cost(self):
-        """Computes the complexity of the given code."""
+    @property
+    def functions(self):
+        """Returns declared functions."""
+        return self._functions
+
+    @property
+    def costs(self):
+        """Returns costs of declared functions."""
+        return self._costs
+
+    @property
+    def shapes(self):
+        """Returns shapes of allocated arrays."""
+        return self._shapes
+
+    @property
+    def mode(self):
+        return self._mode
+
+    @property
+    def visual(self):
+        return self._visual
```

Comment:
What does this do? Can you add a docstring please

Reply:
I couldn't figure out what this is for, it seems like there is no use for it yet

Reply:
Maybe we can remove it? @ratnania ?